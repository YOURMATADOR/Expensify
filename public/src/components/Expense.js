import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

export const Expense = ({ nombre, id, func, descripcion, creadoEl, monto }) => (
  <li>
    <Link to={`/editar/${id}`}>
      <h1> {nombre} </h1>
    </Link>
    <p>{descripcion}</p>
    <p>{moment(creadoEl).format("Do MMM YYYY")}</p>
    <p>{numeral(monto / 100).format("$0,0.00")}</p>
    <button onClick={() => func(id)}>Eliminar</button>
  </li>
);
