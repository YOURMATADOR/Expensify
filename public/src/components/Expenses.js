import React from "react";

import  ListaExpenses  from "./listaExpense";
import  FiltroTexto from "./filtroTexto";
const Expenses = () => (
  <div>
    <FiltroTexto />
    <ListaExpenses />
  </div>
);

export default Expenses;
