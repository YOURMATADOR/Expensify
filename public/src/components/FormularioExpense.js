import "react-dates/initialize";
import React, { Component } from "react";
import { SingleDatePicker } from "react-dates";
import moment from "moment";
//import 'react-dates/lib/css/_datepicker.css'; // ! remover modo test JEST 

import { connect } from "react-redux";
export default class FormularioExpense extends Component {
  constructor(props) {
    super(props);

    this.state = {
      nombre: this.props.expense ? this.props.expense.nombre : "",
      monto: props.expense ? (props.expense.monto / 100).toString() : "1",
      creadoEl: props.expense ? moment(props.expense.creadoEl) : moment(),
      calendarFocus: false,
      error: "",
      descripcion: props.expense ? props.expense.descripcion : ""
    };
  }
  onDateChange = creadoEl => {
    if (creadoEl) {
      this.setState(() => ({ creadoEl }));
    }

  };
  onMontoChange = e => {
    let monto = e.target.value;
    if (!e.target.value || String(e.target.value).match(/^\d{1,}\.?\d{0,2}?$/)) {
      this.setState(() => ({ monto }));
    }
  };
  onFocusChange = ({ focused }) => {
    console.log(focused);
    this.setState(() => ({ calendarFocus: focused }));
  };
  onNameChange = e => {
    let nombre = e.target.value;
    this.setState({ nombre });
  };
  onDescChange = e => {
    let descripcion = e.target.value;
    this.setState(() => ({ descripcion }));
  };
  guardarExpense = e => {
    e.preventDefault();
    if (!this.state.monto || !this.state.nombre) {
      this.setState({ error: "CAMPOS OBLIGATORIOS!" });
    } else {
      const { nombre, monto, creadoEl, descripcion } = this.state;
      this.props.guardar({
        nombre,
        monto: parseFloat(monto, 10) * 100,
        creadoEl: creadoEl.valueOf(),
        descripcion
      });
    }
  };
  render() {
    return (
      <div>
        <form onSubmit={this.guardarExpense}>
          <input
            type="text"
            placeholder={this.state.error ? this.state.error : "Nombre"}
            onChange={this.onNameChange}
            value={this.state.nombre}
          />
          <input
            type="text"
            value={this.state.monto}
            placeholder={this.state.error ? this.state.error : "Monto"}
            onChange={this.onMontoChange}
          />
          <SingleDatePicker
            date={this.state.creadoEl}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocus}
            onFocusChange={this.onFocusChange}
            id="hhhhh"
            small
            numberOfMonths={1}
            isOutsideRange={dia => false}
          />
          <textarea
            placeholder="Descripcion"
            value={this.state.descripcion}
            onChange={this.onDescChange}
          />
          <button>Guardar</button>
        </form>
      </div>
    );
  }
}
// TODO: modificar este componente con redux (hacerlo un componente de presentacion)
