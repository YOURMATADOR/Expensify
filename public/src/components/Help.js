import React from "react";

let Help = props => {
  return (
    <div>
      <h1>Ayuda</h1>
    </div>
  );
};

export default Help;
