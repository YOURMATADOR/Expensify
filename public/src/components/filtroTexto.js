import "react-dates/initialize";
import { DateRangePicker } from "react-dates";
import uuidv1 from "uuid/v1";
import React, { Component } from "react";
import { connect } from "react-redux";
import {
  filtrarTexto,
  filtrarPorFecha,
  filtrarPorMonto,
  cambiarFechaInicio,
  cambiarFechaFin
} from "./../reducer/actions/filtro";
export class FiltroTexto extends Component {
  state = { focusedInput: null };
  onFocusChange = focusedInput => {
    this.setState(() => ({ focusedInput }));
  };
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.texto}
          ref={e => (this.input = e)}
          onChange={e => this.props.func(e.target.value)}
        />
        <select onChange={e => this.props.ordenarPor(e.target.value)}>
          <option value="monto">Monto</option>
          <option value="fecha">Fecha</option>
        </select>
        <DateRangePicker
          startDate={this.props.inicio}
          startDateId={"a"}
          endDate={this.props.fin}
          endDateId={"b"}
          onDatesChange={({ startDate, endDate }) => {
            this.props.actualizaInicio(startDate);
            this.props.actualizaFin(endDate);
          }}
          focusedInput={this.state.focusedInput}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
          small
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  texto: state.Filtro.texto,
  inicio: state.Filtro.inicio,
  fin: state.Filtro.fin
});
const mapDispatchToProps = dispatch => ({
  func: texto => {
    dispatch(filtrarTexto({ texto }));
  },
  ordenarPor: texto => {
    if (texto === "fecha") {
      dispatch(filtrarPorFecha());
    } else if (texto === "monto") {
      dispatch(filtrarPorMonto());
    }
  },
  actualizaInicio: inicio => {
    dispatch(cambiarFechaInicio({ fecha: inicio }));
  },
  actualizaFin: fin => {
    dispatch(cambiarFechaFin({ fecha: fin }));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FiltroTexto);
