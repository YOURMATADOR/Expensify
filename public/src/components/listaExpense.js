import React from "react";
import { connect } from "react-redux";

import { TodosVisibles } from "./../reducer/selectors/visibles";
import { eliminarTodo } from "./../reducer/actions/todos";
import { Expense } from "./Expense";
import { dispatch } from "rxjs/internal/observable/range";
const ListaExpenses = ({ todos, func }) => (
  <div>
    <h1>Expenses</h1>
    <ul>
      {todos.map(i => (
        <Expense key={i.id} {...i} func={func} />
      ))}
    </ul>
  </div>
);

const mapStateToProps = state => ({
  todos: TodosVisibles(state.Todos, state.Filtro)
});
const mapDispatchToProps = dispatch => ({
  func: id => {
    dispatch(eliminarTodo({ id }));
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ListaExpenses);
