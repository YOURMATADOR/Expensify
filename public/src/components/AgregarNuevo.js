import React from "react";
import FormularioExpense from "./FormularioExpense";
import { store } from "./../reducer/store";
import { agregarTodo } from "./../reducer/actions/todos";
const AgregarNuevo = props => {
  return (
    <div>
      <h1>Agregar nuevo Expense</h1>
      <FormularioExpense
        guardar={info => {
          store.dispatch(agregarTodo(info));
          props.history.push("/");
        }}
      />
    </div>
  );
};

export default AgregarNuevo;
