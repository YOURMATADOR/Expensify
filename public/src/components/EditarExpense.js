import React from "react";
import FormularioExpense from "./FormularioExpense";
import {
  editarTodo,
  eliminarTodo,
  editarTodoSync,
  eliminarTodoSync
} from "./../reducer/actions/todos";
import { connect } from "react-redux";
export const EditarExpense = ({ expense, guardar, eliminar }) => (
  <div>
    <h1>Editar Expense</h1>
    <FormularioExpense expense={expense} guardar={guardar} />
    <button onClick={eliminar}>Eliminar</button>
  </div>
);

//? funcionalidad y actualizar con redux

const mapStateToProps = (state, ownProps) => ({
  expense: state.Todos.find(i => i.id === ownProps.match.params.id)
});
const mapDispatchToProps = (dispatch, ownProps) => ({
  guardar: nuevasProps => {
    dispatch(
      editarTodoSync({
        id: ownProps.match.params.id,
        propiedades: { ...nuevasProps }
      })
    );
    ownProps.history.push("/");
  },
  eliminar: () => {
    dispatch(eliminarTodoSync({ id: ownProps.match.params.id }));
    ownProps.history.push("/");
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditarExpense);
