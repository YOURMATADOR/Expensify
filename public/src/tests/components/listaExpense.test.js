import expenses from "../fixtures/expenses";
import { shallow } from "enzyme";
import { ListaExpenses } from "../../components/listaExpense";
import React from "react";

describe("Lista sin propiedades", () => {
  test("Listado con propiedades", () => {
    const wrapper = shallow(<ListaExpenses />);
    expect(wrapper).toMatchSnapshot();
  });
  test('Listado con propiedades', () => {
    const wrapper = shallow(<ListaExpenses todos={expenses}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
