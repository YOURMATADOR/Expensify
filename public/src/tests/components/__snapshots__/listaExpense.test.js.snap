// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lista sin propiedades Listado con propiedades 1`] = `
<div>
  <Connect(SumaTotal) />
  <h1>
    Expenses
  </h1>
  <h1>
    No hay expenses
  </h1>
</div>
`;

exports[`Lista sin propiedades Listado con propiedades 2`] = `
<div>
  <Connect(SumaTotal) />
  <h1>
    Expenses
  </h1>
  <ul>
    <Expense
      creadoEl={"1970-01-08T00:00:00.000Z"}
      descripcion="Algo perron"
      id="1"
      key="1"
      monto={100}
      nombre="Afro"
    />
    <Expense
      creadoEl={"1969-12-25T00:00:00.000Z"}
      descripcion="GG izi"
      id="2"
      key="2"
      monto={200}
      nombre="Poky"
    />
    <Expense
      creadoEl={"1970-01-01T00:00:00.000Z"}
      descripcion="Algo chido"
      id="3"
      key="3"
      monto={300}
      nombre="Tupak"
    />
  </ul>
</div>
`;
